azurerm/internal/services/appplatform/spring_cloud_app_resource.go

```diff
@@ -50,35 +54,76 @@ func resourceArmSpringCloudApp() *schema.Resource {
 				ForceNew:     true,
 				ValidateFunc: validate.SpringCloudServiceName,
 			},
+
+			"identity": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							// other two enum: 'UserAssigned', 'SystemAssignedUserAssigned' are not supported for now
+							ValidateFunc: validation.StringInSlice([]string{
+								string(appplatform.SystemAssigned),
+							}, true),
+							DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
this should be case sensitive - if the API is returning them lower-cased that's an API bug which needs to be fixed